### PR TITLE
feat: add allow_replace to server resource

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 - Add bootvolume_selector to `https://docs.ionos.com/
 - Add servers to `https://docs.ionos.com/`
 - Add cube server and vcpu server to `https://docs.ionos.com/`
+### Enhancement
+- Add `allow_replace` to node `ionoscloud_server` and `ionoscloud_cube_server` resources, which allows the update of immutable server fields will first destroy and then re-create the resource. This field should be used with care, understanding the risks.
```
Comment (adeatcu-ionos) [inline CHANGELOG.md:10]: @cristiGuranIonos 
```suggestion
- Add `allow_replace` to `ionoscloud_server` and `ionoscloud_cube_server` resources, which allows the update of immutable server fields by destroying and then re-creating the resource. This field should be used with care, understanding the risks.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -198,7 +200,7 @@
 - Fix documentation for `ionoscloud_server`, `ionoscloud_volume`, `ionoscloud_lan` resources and `ionoscloud_image` data sources
 
 ### Enhancements
-- Add configurable fields to `ionoscloud_share` resource. Fields that can be set on creation: `description`, `sec_auth_protection`, `licence_type`.
+- Add configurable fields to `ionoscloud_share` resource. Fields that can be set on creation: `description`, `sec_auth_protection`, `licence_type`.
```
Comment (adeatcu-ionos) [inline CHANGELOG.md:203]: @cristiGuranIonos do you use some kind of formatting tool for this file? I'm just curious where these changes are coming from.

File: CHANGELOG.md [hunk 3]
```diff
@@ -643,16 +645,16 @@ To fix, when you try to change immutable fields they will throw an error in the
...
 - Improved lookup in data_sources by using filters
-- Improved tests duration by moving steps from data_source test files in the corresponding resource test files
-- Added workflow to run tests from GitHub actions
+- Improved tests duration by moving steps from data_source test files in the corresponding resource test files 
+- Added workflow to run tests from GitHub actions 
 - Split tests with build tags
 - Improve http client performance and timeouts
 
-### Documentations:
+### Documentations:
```
Comment (adeatcu-ionos) [inline CHANGELOG.md:653]: @cristiGuranIonos these look like some new white spaces, not sure if you actually wanted to add these. Anyway, it doesn't matter.

**Review by adeatcu-ionos — DISMISSED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (cristiGuranIonos) [reply CHANGELOG.md:203]: no idea how they got added back. Maybe on merge in some way. Will leave as is, it does not matter.

**Pull request merged**